### PR TITLE
[Triggers] make message batch size configurable

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerConfig.scala [hunk 1]
```diff
@@ -249,6 +249,16 @@ object RunnerConfig {
...
       }
 
+    opt[Long]("max-batch-size")
+      .optional()
+      .text(
+        s"maximal number of message processed between two high-level rule triggers. Defaults to ${DefaultTriggerRunnerConfig.maximumBatchSize}"
+      )
+      .action((size, cli) =>
+        if (size > 0) cli.copy(triggerConfig = cli.triggerConfig.copy(maximumBatchSize = 1))
+        else throw new IllegalArgumentException(s"batch size must be strictlty positive")
```
Comment (carlpulley-da) [inline triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerConfig.scala:259]: `strictlty` -> `strictly`

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerConfig.scala [hunk 2]
```diff
@@ -249,6 +249,16 @@ object RunnerConfig {
           throw new IllegalArgumentException(s"Unsupported logging encoder $other")
       }
 
+    opt[Long]("max-batch-size")
+      .optional()
+      .text(
+        s"maximal number of message processed between two high-level rule triggers. Defaults to ${DefaultTriggerRunnerConfig.maximumBatchSize}"
```
Comment (carlpulley-da) [inline triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerConfig.scala:255]: Nit pick: maybe write "maximum number of messages..." instead?

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerConfig.scala [hunk 3]
```diff
@@ -249,6 +249,16 @@ object RunnerConfig {
...
+      )
+      .action((size, cli) =>
+        if (size > 0) cli.copy(triggerConfig = cli.triggerConfig.copy(maximumBatchSize = 1))
```
Comment (carlpulley-da) [inline triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerConfig.scala:258]: Here (and in `Cli.scala` below), should this be ```if (size > 0) cli.copy(triggerConfig = cli.triggerConfig.copy(maximumBatchSize = size))```?

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Cli.scala [hunk 4]
```diff
@@ -323,10 +323,19 @@ private[trigger] object Cli {
...
         case ("plain", cli) => cli.copy(logEncoder = LogEncoder.Plain)
-        case (other, _) =>
-          throw new IllegalArgumentException(s"Unsupported logging encoder $other")
+        case (other, _) => throw new IllegalArgumentException(s"Unsupported logging encoder $other")
       }
 
+    opt[Long]("max-batch-size")
+      .optional()
+      .text(
+        s"maximal number of message processed between two high-level rule triggers. Defaults to ${DefaultTriggerRunnerConfig.maximumBatchSize}"
```
Comment (carlpulley-da) [inline triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Cli.scala:332]: Nit pick: maybe write "maximum number of messages..." instead?

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Cli.scala [hunk 5]
```diff
@@ -323,10 +323,19 @@ private[trigger] object Cli {
...
+      )
+      .action((size, cli) =>
+        if (size > 0) cli.copy(triggerConfig = cli.triggerConfig.copy(maximumBatchSize = 1))
+        else throw new IllegalArgumentException(s"batch size must be strictlty positive")
```
Comment (carlpulley-da) [inline triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Cli.scala:336]: `strictlty` -> `strictly`

**Review by carlpulley-da — APPROVED** [review]
Other than a small comment, LGTM!

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerConfig.scala (hunk 3 above)
Comment (remyhaemmerle-da) [reply triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerConfig.scala:258]: good catch.

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerConfig.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerConfig.scala:259]: :+1

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerConfig.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerConfig.scala]: :+1:

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Cli.scala (hunk 4 above)
Comment (remyhaemmerle-da) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Cli.scala]: :+1:

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Cli.scala (hunk 5 above)
Comment (remyhaemmerle-da) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Cli.scala]: :+1:

**Review by carlpulley-da — APPROVED** [review]

**Pull request merged**